Update Ubuntu to 18.04 and PHP to 7.2

filips123 opened the pull request:
### What does this PR do?

It updates Ubuntu from 16.04 to 18.04. It also updates PHP from 7.0 to 7.2.

### What issues does this PR fix or reference?

Fixes #184
Fixes #193
Fixes codenvy/codenvy#2673

### Tests written?

No

### Docs updated?

No

Siranix commented:
@filips123 I don't think they would be able to just push this as the changes you're doing are probably breaking other stacks.

What you're doing is that you're updating Ubuntu to 18.04, meaning you'd have to test and adjust the other stacks to also match and work on Ubuntu 18.04.

Just a note:
This PHP7.2 change doesn't work in combination with Ubuntu 16.04 as the base stack.

filips123 commented:
Yes, other stacks should be tested and fixed with Ubuntu 18.04.

To use PHP 7.2 in Ubuntu 16.034, you need to add PPA repository (https://thishosting.rocks/install-php-on-ubuntu/#install-php-7-2-ubuntu).

Siranix commented:
@filips123 Well aware of that, but what I mean to say is that you'd have to start fixing the other stacks as well on the PR in order to get it accepted etc.

I found the best way to run 7.2 in Eclipse Che by running a basic PHP Workspace and then upgrading the PHP to 7.2 and adjust the apache2. I followed somewhat the same path as the article you provided.

filips123 commented:
@Siranix Is there a way to quickly test all stacks?

Siranix commented:
I think the only real way is to run every stack with their dedicated test
(some have no test). Sorry, I'm not a Che expert, was just checking if this
ever got fixed.

Especially watch for dockerfiles that have links in them as you'd have to
update those 90% of the time.

On Fri, Oct 5, 2018, 5:05 PM Filip Š <notifications@github.com> wrote:

> @Siranix <https://github.com/Siranix> Is there a way to quickly test all
> stacks?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/eclipse/che-dockerfiles/pull/214#issuecomment-427397491>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AULZ87rl303ZQjXIWk5rw5d9wE2tBsm3ks5uh3VNgaJpZM4XEKuu>
> .
>

filips123 commented:
How should I test them? Is building the images enough? If it is I will probably modify [`build.sh`](https://github.com/eclipse/che-dockerfiles/blob/master/build.sh) script to build all Ubuntu based stacks.

filips123 commented:
I figure out that there should be `DEBIAN_FRONTEND=noninteractive` before `apt-get update` and `apt-get install` because system currently prompt user for timezone data.

I will add it tomorrow.

filips123 commented:
I fixed stacks and now they build successfully with Ubuntu 18.04.
I tested this with [`build.sh`](https://github.com/eclipse/che-dockerfiles/blob/master/build.sh) script.

I replaced Python installation from TAR with APT because it not works.
